homeassistant/components/simplisafe/__init__.py

```diff
@@ -231,9 +243,33 @@ def _async_register_base_station(
     )
 
 
-async def async_setup_entry(  # noqa: C901
-    hass: HomeAssistant, entry: ConfigEntry
-) -> bool:
+@callback
+def _async_get_system_for_service_call(
+    hass: HomeAssistant, call: ServiceCall
+) -> SystemV2 | SystemV3:
+    """Get the SimpliSafe system related to a service call (by device ID)."""
+    device_id = call.data[ATTR_DEVICE_ID]
+    device_registry = dr.async_get(hass)
+
+    if alarm_control_panel_device_entry := device_registry.async_get(device_id):
```

Comment:
you can simplify the code
```suggestion
    if (alarm_control_panel_device_entry := device_registry.async_get(device_id)) is None:
        raise ValueError(f"No system for device ID: {device_id}")
```

Reply:
I also think that you can probably raise `vol.Invalid` since an invalid device ID was specified